app/src/main/java/openfoodfacts/github/scrachx/openfood/models/FoodAPIRestClientUsage.java

```diff
@@ -277,21 +276,14 @@ public void onStart() {
 
             @Override
             public void onSuccess(int statusCode, cz.msebera.android.httpclient.Header[] headers, byte[] responseBody) {
-                // called when response HTTP status is "200 OK"
                 try {
-                    ObjectMapper objectMapper = new ObjectMapper();
-                    try {
-                        JsonNode root = objectMapper.readTree(responseBody);
-                        List<Allergen> la = objectMapper.readValue(root.path("tags").toString(), new TypeReference<List<Allergen>>() {});
-                        for (Allergen a : la) {
-                            Allergen al = new Allergen(a.getUrl(),a.getName(),a.getProducts(),a.getIdAllergen());
-                            al.save();
-                        }
-                        onAllergensCallback.onAllergensResponse(true);
-                    } catch (InvalidFormatException e) {
-                        e.printStackTrace();
-                        onAllergensCallback.onAllergensResponse(false);
+                    AllergenRestResponse restResponse = JsonUtils.readFor(AllergenRestResponse.class)
+                            .readValue(responseBody);
+
+                    for (Allergen allergen : restResponse.getAllergens()) {
+                        allergen.save();
```

Comment:
It's to store in the DB all allergens from the API, and then I can use it in this view : AlertUserFragment